File: Selenium/Activities/Activity14.java
package examples;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity14 {

    public static void main(String[] args) {

        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/tables");

        System.out.println("Page Title: " + driver.getTitle());

        List<WebElement> cols = driver.findElements(
                By.xpath("//table[contains(@class,'table-auto')]//thead//th"));
        System.out.println("Number of Columns: " + cols.size());

        List<WebElement> rows = driver.findElements(
                By.xpath("//table[contains(@class,'table-auto')]//tbody//tr"));
        System.out.println("Number of Rows: " + rows.size());

        WebElement bookNameBefore = driver.findElement(
                By.xpath("//table[contains(@class,'table-auto')]//tbody//tr[5]//td[2]"));
        System.out.println("Book Name before sorting: " + bookNameBefore.getText());

        driver.findElement(
                By.xpath("//table[contains(@class,'table-auto')]//thead//th[5]")).click();

        WebElement bookNameAfter = driver.findElement(
                By.xpath("//table[contains(@class,'table-auto')]//tbody//tr[5]//td[2]"));
        System.out.println("Book Name after sorting: " + bookNameAfter.getText());

        driver.quit();
    }
}

File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/tables")

print("Page title:", driver.title)

cols = driver.find_elements(By.XPATH, "//table[contains(@class,'table-auto')]//thead//th")
print("Number of columns:", len(cols))

rows = driver.find_elements(By.XPATH, "//table[contains(@class,'table-auto')]//tbody//tr")
print("Number of rows:", len(rows))

book_before = driver.find_element(
    By.XPATH, "//table[contains(@class,'table-auto')]//tbody//tr[5]//td[2]"
)
print("Book name before sorting:", book_before.text)

driver.find_element(
    By.XPATH, "//table[contains(@class,'table-auto')]//thead//th[5]"
).click()

time.sleep(2)

book_after = driver.find_element(
    By.XPATH, "//table[contains(@class,'table-auto')]//tbody//tr[5]//td[2]"
)
print("Book name after sorting:", book_after.text)

driver.quit()